Source/JavaScriptCore/assembler/AssemblerBuffer.h

```diff
@@ -232,6 +241,24 @@ namespace JSC {
             deallocatePinForCurrentThread();
         }
 
+        ALWAYS_INLINE uint32_t update(uint32_t instruction, uint32_t byteIdx)
+        {
+            uint32_t instrIdx = byteIdx / sizeof(instruction);
```

Comment:
Would this penalize ARM64 perf for the sake of x64 code?  Maybe the compiler will optimize it out, but can you check?  This is in a perf sensitive area of code.

Reply:
The two places we call `update` were previously:

AssemblerBuffer.h:463:
```
uint32_t hash = m_hash.update(value, m_index / sizeof(IntegralType));
```

LinkBuffer.h:323:
```
unsigned index = (bitwise_cast<uint8_t*>(ptr) - inData) / 4;
uint32_t hash = verifyUncompactedHash.update(value, index);
```

Since this function is `ALWAYS_INLINE` the resulting assembly is the same before and after:
```
JavaScriptCore`void JSC::LinkBuffer::copyCompactAndLinkCode<unsigned int>(JSC::MacroAssembler&, JSC::JITCompilationEffort)::'lambda'(unsigned int const*)::operator()(unsigned int const*) const:
    0x10b009404 <+0>:  pacibsp 
    0x10b009408 <+4>:  sub    sp, sp, #0x40
    0x10b00940c <+8>:  stp    x29, x30, [sp, #0x30]
    0x10b009410 <+12>: add    x29, sp, #0x30
    0x10b009414 <+16>: stur   x0, [x29, #-0x8]
    0x10b009418 <+20>: stur   x1, [x29, #-0x10]
    0x10b00941c <+24>: ldur   x8, [x29, #-0x8]
    0x10b009420 <+28>: str    x8, [sp, #0x8]
```

I'll still run an A/B test on the final changes before/after to be sure, of course.